File: backend/app/models/agent.py
"""Agent data models."""

from typing import List, Optional, Any
from pydantic import BaseModel


class MessagePart(BaseModel):
    """Message part model."""
    text: str


class NewMessage(BaseModel):
    """New message model."""
    parts: List[MessagePart]
    role: str = "user"


class AgentRequest(BaseModel):
    """Agent request model."""
    appName: str = "agent"
    userId: str = "user"
    sessionId: Optional[str] = None
    newMessage: NewMessage
    streaming: bool = False


class AgentMessageRequest(BaseModel):
    """Agent message request from frontend."""
    message: str
    sessionId: Optional[str] = None


class AgentResponse(BaseModel):
    """Agent response model."""
    success: bool
    response: Any
    error: Optional[str] = None


class AgentStreamRequest(BaseModel):
    """Agent stream request from frontend."""
    message: str
    sessionId: Optional[str] = None


class SessionCreateRequest(BaseModel):
    """Session creation request."""
    additionalProp1: dict = {}


class SessionCreateResponse(BaseModel):
    """Session creation response."""
    sessionId: str
    success: bool
    error: Optional[str] = None

File: frontend/src/types/agent.ts
export interface AgentMessage {
  id: string;
  text: string;
  role: 'user' | 'agent';
  timestamp: number;
}

export interface AgentMessageRequest {
  message: string;
  sessionId?: string;
}

export interface AgentResponse {
  success: boolean;
  response: any;
  error?: string;
}

export interface AgentStreamRequest {
  message: string;
  sessionId?: string;
}

export interface AgentStreamResponse {
  text?: string;
  done?: boolean;
  error?: string;
}
